utility/src/main/java/org/oppia/android/util/logging/CurrentAppScreenNameIntentDecorator.kt

```diff
@@ -0,0 +1,33 @@
+package org.oppia.android.util.logging
+
+import android.content.Intent
+import org.oppia.android.app.model.CurrentAppScreen
+import org.oppia.android.app.model.ScreenName
+import org.oppia.android.util.extensions.getProtoExtra
+import org.oppia.android.util.extensions.putProtoExtra
+
+private var CURRENT_APP_SCREEN_NAME_KEY = "CURRENT_APP_SCREEN_NAME_KEY"
+
+/** Utility to provide utilities related to wrapping the current app-screen name in an intent. */
+object CurrentAppScreenNameIntentDecorator {
+  /**
+   * Packs the intent with a [CurrentAppScreen] proto object that sets [screenName] as the current
+   * screen.
+   */
+  fun Intent.decorateWithScreenName(screenName: ScreenName?) {
+    if (screenName != null) {
+      this.putProtoExtra(
+        CURRENT_APP_SCREEN_NAME_KEY,
+        CurrentAppScreen.newBuilder().setScreenName(screenName).build()
+      )
+    }
+  }
+
+  /** Returns [ScreenName] after unpacking intent. */
```

Comment:
```suggestion
  /** Returns the [ScreenName] packed in the intent via [decorateWithScreenName]. */
```

Also for clarity since it's nice to tie the read & write methods together in documentation.

Reply:
done.